iroha/src/account.rs

```diff
@@ -5,70 +5,149 @@ use std::collections::BTreeMap;
 #[derive(Debug, Clone, Encode, Decode)]
 pub struct Account {
     pub id: Id,
-    pub assets: BTreeMap<Id, Asset>,
+    pub assets: BTreeMap<<Asset as Identifiable>::Id, Asset>,
+    signatories: Vec<PublicKey>,
 }
 
 impl Account {
-    pub fn new(account_id: Id) -> Self {
+    pub fn new(account_name: &str, container: &str, public_key: PublicKey) -> Self {
         Account {
-            id: account_id,
+            id: Id::new(account_name, container),
             assets: BTreeMap::new(),
+            signatories: vec![public_key],
         }
     }
 }
 
+/// Identification of an Account. Consists of Account's name and Domain's name.
+///
+/// # Example
+///
+/// ```
+/// use iroha::account::Id;
+///
+/// let id = Id::new("user", "company");
+/// ```
+#[derive(Clone, Debug, PartialEq, PartialOrd, Ord, Eq, std::hash::Hash, Encode, Decode)]
+pub struct Id {
+    pub name: String,
+    pub container: String,
+}
+
+impl Id {
+    pub fn new(name: &str, container: &str) -> Self {
+        Id {
+            name: name.to_string(),
+            container: container.to_string(),
+        }
+    }
+}
+
+impl From<&str> for Id {
+    fn from(string: &str) -> Id {
+        let vector: Vec<&str> = string.split('@').collect();
+        Id {
+            name: String::from(vector[0]),
+            container: String::from(vector[1]),
```

Comment:
accounts are namespaces by domain, e.g., makoto@sora, so this "container" for an account will always be a domain.